hpobench/benchmarks/ml_mmfb/histgb_benchmark.py

```diff
@@ -0,0 +1,136 @@
+
+    @staticmethod
+    def get_configuration_space(seed: Union[int, None] = None) -> CS.ConfigurationSpace:
+        """Parameter space to be optimized --- contains the hyperparameters"""
+        cs = CS.ConfigurationSpace(seed=seed)
+
+        cs.add_hyperparameters([
+            CS.UniformIntegerHyperparameter(
+                'max_depth', lower=6, upper=30, default_value=6, log=True
+            ),
+            # TODO: The parameter max_leaf_node is not accepted. Changed it from max_leaf_node to max_leaf_nodes
+            CS.UniformIntegerHyperparameter(
+                'max_leaf_nodes', lower=2, upper=64, default_value=32, log=True
+            ),
+            # TODO: The parameter eta is not accepted. Do you mean learning_rate? Changed it from eta to learning_rate
+            CS.UniformFloatHyperparameter(
+                'learning_rate', lower=2**-10, upper=1, default_value=0.1, log=True
+            ),
+            CS.UniformFloatHyperparameter(
+                'l2_regularization', lower=2**-10, upper=2**10, default_value=0.1, log=True
+            )
+        ])
+        return cs
+
+    @staticmethod
+    def get_fidelity_space(seed: Union[int, None] = None) -> CS.ConfigurationSpace:
+        """Fidelity space available --- specifies the fidelity dimensions
+
+        If fidelity_choice is 0
+            Fidelity space is the maximal fidelity, akin to a black-box function
+        If fidelity_choice is 1
+            Fidelity space is a single fidelity, in this case the number of trees (n_estimators)
+        If fidelity_choice is 2
+            Fidelity space is a single fidelity, in this case the fraction of dataset (subsample)
+        If fidelity_choice is >2
+            Fidelity space is multi-multi fidelity, all possible fidelities
+        """
+        raise NotImplementedError()
+
+    @staticmethod
+    def _get_fidelity_choices(ntrees_choice: str, subsample_choice: str) -> Tuple[Hyperparameter, Hyperparameter]:
+
+        assert ntrees_choice in ['fixed', 'variable']
+        assert subsample_choice in ['fixed', 'variable']
+
+        fidelity1 = dict(
+            # TODO: this value was 100 in the original code. Please check if 100 or 1000.
+            fixed=CS.Constant('n_estimators', value=1000),
+            variable=CS.UniformIntegerHyperparameter(
+                'n_estimators', lower=100, upper=1000, default_value=1000, log=False
+            )
+        )
+        fidelity2 = dict(
+            fixed=CS.Constant('subsample', value=1),
+            variable=CS.UniformFloatHyperparameter(
+                'subsample', lower=0.1, upper=1, default_value=1, log=False
+            )
+        )
+        ntrees = fidelity1[ntrees_choice]
+        subsample = fidelity2[subsample_choice]
+        return ntrees, subsample
+
+    def init_model(self, config: Dict, fidelity: Dict = None, rng: Union[int, np.random.RandomState, None] = None):
```

Comment:
Why the `Dict` enforcing of config and fidelity? 
I would rather have it be flexible and take both ConfigSpace and Dict if Dict needs to be supported explicitly

Reply:
Alright. I'll change it. However, the wrapper in the objective function converts the `CS.Configuration` to `Dict` in the last step. So basically, I have changed it after it crashed. It was called with a Dict and then the Dict does not have a .get_dictionary() function. 

Is then something like a check in the beginning of the function fine:
``` 
isinstance(config, CS.Configuration):
    config = config.get_dictionary()
```